recipes/vtk/all/conanfile.py

```diff
@@ -0,0 +1,1449 @@
+# RECIPE MAINTAINER NOTES:
+# - Read vtk's Documentation/release/9.1.md for important notes about versions and forks
+# - Also read vtk's Documentation/build.md for information about build settings and flags
+# - Modify build_requirements() to match the version of CMake that VTK tested with that release.
+
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration, ConanException
+from conan.tools.cmake import CMakeToolchain, CMakeDeps, CMake, cmake_layout
+from conan.tools.build import check_min_cppstd
+from conan.tools.microsoft import check_min_vs, is_msvc
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy, rmdir, save, rename, collect_libs, load, replace_in_file
+from conan.tools.scm import Version
+
+from shutil import which
+
+import os
+import textwrap
+
+import json # for auto-component generation
+import itertools # for autoinit header generation
+
+# Enable to keep VTK-generated cmake files, to check contents
+_debug_packaging = False
+
+required_conan_version = ">=1.59.0"
+
+
+# for self.options.group_* and self.options.
+
+# https://vtk.org/doc/nightly/html/group__module.html
+# QUOTE #
+# QUOTE # Modules and groups are enable and disable preferences are specified using a 5-way flag setting:
+# QUOTE #
+# QUOTE # YES: The module or group must be built.
+# QUOTE # NO: The module or group must not be built.
+# QUOTE # WANT: The module or group should be built if possible.
+# QUOTE # DONT_WANT: The module or group should only be built if required (e.g., via a dependency).
+# QUOTE # DEFAULT: Acts as either WANT or DONT_WANT based on the group settings for the module or WANT_BY_DEFAULT option to vtk_module_scan if no other preference is specified. This is usually handled via another setting in the main project.
+# QUOTE # If a YES module preference requires a module with a NO preference, an error is raised.
+# QUOTE #
+# QUOTE # A module with a setting of DEFAULT will look for its first non-DEFAULT group setting and only if all of those are set to DEFAULT is the WANT_BY_DEFAULT setting used.
+
+# This means we don't truly know what is required until after configuration,
+# and we can't configure without the correct requires!
+# So we will just do our best and the consumer can adjust as required.
+
+# See notes below in _is_module_enabled()
+
+def _is_module_YES_or_WANT(flag):
+    return flag in ["YES", "WANT"]
+
+def _is_module_NO(flag):
+    return flag in ["NO"]
+
+
+
+# Load the entire list of modules
+# Discovered by this command in src folder:
+# find . -name "vtk.module" -exec head -2 '{}' \; | grep -v ^NAME$ | grep "^  VTK::" | cut -c8- | sort | uniq > vtk-modules.txt
+# TODO put these in an external file... for now, using vim to bring it into the recipe by hand
+vtk_all_modules = [
```

Comment:
This is currently a duplicate of the content of the `vtk-modules.txt` file, right?

Reply:
You could use https://docs.conan.io/1/reference/conanfile/tools/files/basic.html#conan-tools-files-load + some string splitting to get a list

Reply:
Correct, however, vtk-modules.txt was a recent dump to check for differences.
I would like to make it import the file, but, need to also understand that the modules come from the source, and they could change with each version!
So currently I have to extract the source, grep for modules and populate the list.
I guess that also means there should be a list per version too?

Reply:
Yes, sounds like there should be a list per version. Have a look at the Qt recipe that does a similar thing https://github.com/conan-io/conan-center-index/tree/master/recipes/qt/6.x.x